Xamarin.Forms.Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -970,7 +977,10 @@ static IEnumerable<Instruction> Set(VariableDefinition parent, string localName,
 					yield return Instruction.Create(OpCodes.Unbox_Any, module.Import(propertyType));
 				else if (vardef.VariableType.IsValueType && propertyType.FullName == "System.Object")
 					yield return Instruction.Create(OpCodes.Box, vardef.VariableType);
-				yield return Instruction.Create(OpCodes.Callvirt, propertySetterRef);
+				if (parent.VariableType.IsValueType)
+					yield return Instruction.Create(OpCodes.Call, propertySetterRef);
+				else
+					yield return Instruction.Create(OpCodes.Callvirt, propertySetterRef);
```

Comment:
setting properties on valuetypes require to load their address (`ldloca`) and doesn't support virtual calls.